cognite/neat/graph/queries/_base.py

```diff
@@ -119,21 +124,52 @@ def describe(
                 "null",
             ]:
                 continue
-            # we are skipping deep validation with Pydantic to remove namespace here
-            # as it reduces time to process triples by 10-15x
-            value = remove_namespace_from_uri(object_, validation="prefix")
-
-            # use-case: calling describe without renaming properties
-            # losing the namespace from the predicate!
-            if not property_renaming_config and predicate != RDF.type:
-                property_values[remove_namespace_from_uri(predicate, validation="prefix")].append(value)
-            elif predicate == RDF.type:
-                property_values[RDF.type].append(value)
-            # use-case: calling describe with renaming properties
-            # renaming the property to the new name, if the property is defined
-            # in the RULES sheet
-            elif property_renaming_config and (property_ := property_renaming_config.get(predicate, None)):
+
+            # set property
+            if property_renaming_config and predicate != RDF.type:
+                property_ = property_renaming_config.get(
+                    predicate, remove_namespace_from_uri(predicate, validation="prefix")
+                )
+            elif not property_renaming_config and predicate != RDF.type:
+                property_ = remove_namespace_from_uri(predicate, validation="prefix")
+            else:
+                property_ = RDF.type
+
+            # set value
+            # if it is URIRef and property type is object property, we need to remove namespace
+            # if it URIref but we are doing this into data type property, we do not remove namespace
+            # case 1 for RDF type we remove namespace
+            if property_ == RDF.type:
+                value = remove_namespace_from_uri(object_, validation="prefix")
+
+            # case 2 for define object properties we remove namespace
+            elif (
+                isinstance(object_, URIRef)
+                and property_types
+                and property_types.get(property_, None) == EntityTypes.object_property
+            ):
+                value = remove_namespace_from_uri(object_, validation="prefix")
+
+            # case 3 when property type is not defined and returned value is URIRef we remove namespace
+            elif isinstance(object_, URIRef) and not property_types:
+                value = remove_namespace_from_uri(object_, validation="prefix")
+
+            # case 4 for data type properties we do not remove namespace but keep the entire value
+            # but we drop the datatype part, and keep everything to be string (data loader will do the conversion)
+            # for value type it expects (if possible)
+            else:
+                value = str(object_)
```

Comment:
If I understand it correctly, this is the main part of change? We no longer remove namespace from data types?